packages/utils/src/bulk-query-patients.ts

```diff
@@ -0,0 +1,124 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+// keep that ^ on top
+import { ConsolidatedCountResponse, MetriportMedicalApi, Patient } from "@metriport/api-sdk";
+import fs from "fs";
+import { getEnvVar, getEnvVarOrFail } from "./shared/env";
+
+// auth stuff
+const apiKey = getEnvVarOrFail("API_KEY");
+const facilityId = getEnvVarOrFail("FACILITY_ID");
+const apiUrl = getEnvVarOrFail("API_URL");
+const metriportAPI = new MetriportMedicalApi(apiKey, {
+  baseAddress: apiUrl,
+});
+
+// query stuff
+const delayTime = parseInt(getEnvVar("BULK_QUERY_DELAY_TIME") ?? "5000");
+const patientChunkSize = parseInt(getEnvVar("PATIENT_CHUNK_SIZE") ?? "25");
+const patientChunkDelayJitterMs = parseInt(getEnvVar("PATIENT_CHUNK_DELAY_JITTER_MS") ?? "1000");
+const queryPollDurationMs = 10_000;
+const maxQueryDurationMs = 71_000; // CW has a 70s timeout, so this is the maximum duration any doc query can take
+const maxDocQueryAttemts = 3;
+
+// csv stuff
+const cxName = getEnvVarOrFail("CX_NAME");
+const csvHeader =
+  "patientId,firstName,lastName,state,queryAttemptCount,docCount,fhirResourceCount,fhirResourceDetails,status\n";
+const curDateTime = new Date();
+const csvName = `./${replaceAll(cxName, " ", "").trim()}-DocQuery-${curDateTime.toISOString()}.csv`;
+
+function replaceAll(string: string, search: string, replace: string): string {
+  return string.split(search).join(replace);
+}
+
+function initCsv() {
+  fs.writeFileSync(csvName, csvHeader);
+}
+
+function sleep(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function queryDocsForPatient(patient: Patient) {
+  let docQueryAttempts = 0;
+  let docCount = 0;
+  let fhirResourceCount = 0;
+  let status: "completed" | "docs-not-found" = "docs-not-found";
+  let queryComplete = false;
+  let fhirResourceCounts: ConsolidatedCountResponse = {
+    filter: { resources: "" },
+    resources: {},
+    total: 0,
+  };
+  while (docQueryAttempts < maxDocQueryAttemts) {
+    console.log(`>>> Starting doc query for patient ${patient.id}...`);
+    await metriportAPI.startDocumentQuery(patient.id, patient.facilityIds[0]);
+    // add a bit of jitter to the requests
+    await sleep(200 + Math.random() * patientChunkDelayJitterMs);
+    const queryStartTime = Date.now();
+    while (Date.now() - queryStartTime < maxQueryDurationMs) {
+      const docQueryStatus = await metriportAPI.getDocumentQueryStatus(patient.id);
+      // ensure at least 1 doc was downloaded, and both the download + conversion have completed
+      if (
+        docQueryStatus.download &&
+        docQueryStatus.download.status === "completed" &&
+        docQueryStatus.download.successful &&
+        docQueryStatus.download.successful > 0 &&
```

Comment:
Why looking at successful, wouldn't `status !== processing` suffice?

Reply:
This is purposefully here to trigger a retry if we initially get 0 docs - due to the behavior we've previously seen